### ISSUE TITLE
Select query in leftJoin selecting all (*) columns is not mapping custom column names ON join

### ISSUE THREAD

**Issue by Creatium** [typeorm/typeorm#10629]
### Issue description

Column names are not being mapped properly in leftJoin "ON" part

### Expected Behavior

Column names should be mapped.

### Actual Behavior

I am doing a `leftJoin` with sub query and in that sub query I have `.select("table.*")` statement, it allows me to just select all the columns from one table. Then on the `ON` part of this join I am referencing one of the selected columns to match. And specifically in this part where the custom column names are not being mapped properly.

### Steps to reproduce

A sample query:

```
MySource.getRepository(TableOneModel)
  .createQueryBuilder("tableOne")
  .select("tableOne.someColumn")
  .leftJoin((subQuery) => {
     subQuery
      .select("tableTwo.*")
      .from(TableTwoModel, "tableTwo");
     
      return subQuery;
     },
     "tbTwo", 
     "tbTwo.customColumn = tableOne.someColumn"
  )
  .getRawMany();

```

The part that is not mapped properly is: tbTwo.**customColumn** = tableOne.someColumn.

If I do `getSql()` function to retrieve the generated query, the `customColumn` name is there as in code, while it should instead be `custom_column` as is stated in model.

This results in error `Unknown column 'tbTwo.customColumn' in 'field list'`.

If I add `.addSelect("tableTwo.customColumn", "customColumn")` in sub query then everything works fine. But I should not be required to do this, as I am telling query to select all columns. And in my model I do not have "no select" option specified. 

Now maybe my way of doing this `leftJoin` is wrong? It works perfectly otherwise.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | 18.16.1  |
| Typescript version  | 5.2.2  |
| TypeORM version     | 0.3.17  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [X] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by nirbhaykumar1999** [comment]
When we are using `select("tableTwo.*")`, it actually returns the datat in format: `tableTwo_custom_column`
While `.addSelect("tableTwo.customColumn", "customColumn")` returns `customColumn`
This is not a bug but expected behaviour

**Comment by Creatium** [comment]
> When we are using `select("tableTwo.*")`, it actually returns the datat in format: `tableTwo_custom_column` While `.addSelect("tableTwo.customColumn", "customColumn")` returns `customColumn` This is not a bug but expected behaviour

Yes, this is what happens. Not sure why it is expected behaviour, can you elaborate? Because now it requires me to either mention all the columns or use raw column names in queries, which kind of defeats the purpose of name wrapping.
